mcsema/BC/Function.cpp

```diff
@@ -452,6 +452,29 @@ static void CreateLandingPad(TranslationContext &ctx,
     args[0] = ir.CreateLoad(ctx.cfg_func->stack_ptr_var, true);
     args[1] = ir.CreateLoad(ctx.cfg_func->frame_ptr_var, true);
     args[2] = ir.CreateLoad(var_value, true);
+#endif
+    auto handler = GetExceptionHandlerPrologue();
+    ir.CreateCall(handler, args);
+  } else {
+    auto get_index_func = gModule->getFunction("__mcsema_get_type_index");
+    if (!get_index_func) {
+      get_index_func = llvm::Function::Create(
+          llvm::FunctionType::get(llvm::Type::getInt64Ty(*gContext), false),
+          llvm::GlobalValue::ExternalWeakLinkage,
+          "__mcsema_get_type_index", gModule);
+    }
+    auto bp_var = ir.CreateCall(get_index_func);
```

Comment:
IS this related to the base pointer???

Reply:
I am using it to read the `RDX` register. It has the associated type index. Is there a good way to read the value of the registers?

Reply:
What I mean is the naming of `bp_var`. Also, if this is reading RDX, then what you're saying is that it is reading something register/address-sized, and so you shouldn't be hardcoding it as a 64-bit int type, as on a 32-bit system, it should return a 32-bit value. Almost everywhere you use 64-bit ints, you should instead be using `gWordType` [1].

[1] https://github.com/trailofbits/mcsema/blob/master/mcsema/BC/Util.h#L44